traffic_ops/client/stats_summary.go

```diff
@@ -15,111 +15,69 @@ package client
 import (
 	"encoding/json"
 	"fmt"
-	"strconv"
+	"net"
+	"net/url"
 
 	tc "github.com/apache/trafficcontrol/lib/go-tc"
 )
 
-// SummaryStats ...
-// Deprecated: use GetSummaryStats
-func (to *Session) SummaryStats(cdn string, deliveryService string, statName string) ([]tc.StatsSummary, error) {
-	ss, _, err := to.GetSummaryStats(cdn, deliveryService, statName)
-	return ss, err
-}
+const (
+	basePathSummaryStats = apiBase + "/stats_summary"
+)
 
-func (to *Session) GetSummaryStats(cdn string, deliveryService string, statName string) ([]tc.StatsSummary, ReqInf, error) {
-	var queryParams []string
-	if len(cdn) > 0 {
-		queryParams = append(queryParams, fmt.Sprintf("cdnName=%s", cdn))
-	}
-	if len(deliveryService) > 0 {
-		queryParams = append(queryParams, fmt.Sprintf("deliveryServiceName=%s", deliveryService))
+// GetSummaryStats gets a list of summary stats with the ability to filter on cdn,deliveryService and/or stat
+func (to *Session) GetSummaryStats(cdn, deliveryService, statName *string) (tc.StatsSummaryResponse, ReqInf, error) {
+	resp := tc.StatsSummaryResponse{}
+
+	param := url.Values{}
+	if cdn != nil {
+		param.Add("cdnName", *cdn)
 	}
-	if len(statName) > 0 {
-		queryParams = append(queryParams, fmt.Sprintf("statName=%s", statName))
+	if deliveryService != nil {
+		param.Add("deliveryServiceName", *deliveryService)
 	}
-	queryURL := "/api/1.2/stats_summary.json"
-	queryParamString := "?"
-	if len(queryParams) > 0 {
-		for i, param := range queryParams {
-			if i == 0 {
-				queryParamString += param
-			} else {
-				queryParamString += fmt.Sprintf("&%s", param)
-			}
-		}
-		queryURL += queryParamString
+	if statName != nil {
+		param.Add("statName", *statName)
 	}
 
-	resp, remoteAddr, err := to.request("GET", queryURL, nil)
-	reqInf := ReqInf{CacheHitStatus: CacheHitStatusMiss, RemoteAddr: remoteAddr}
-	if err != nil {
-		return nil, reqInf, err
+	url := basePathSummaryStats
+	if len(param) > 0 {
+		url = fmt.Sprintf("%s?%s", basePathSummaryStats, param.Encode())
 	}
-	defer resp.Body.Close()
-
-	var data tc.StatsSummaryResponse
-	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
-		return nil, reqInf, err
+	reqInf, err := get(to, url, &resp)
+	if err != nil {
+		return resp, reqInf, err
 	}
-
-	return data.Response, reqInf, nil
+	return resp, reqInf, nil
```

Comment:
Why bother with this `if` block? You could just `return resp, reqInf, err`. If `err` is `nil` then it'll be `nil`; no need to check.